app/controllers/application_controller/filter.rb

```diff
@@ -307,11 +307,11 @@ def adv_search_button
             {:model => ui_lookup(:model => @edit[@expkey][:exp_model]),
              :name => @edit[:new_search_name]})
           @edit[@expkey].select_filter(s)
-          @edit[:new_search_name] = @edit[:adv_search_name] = @edit[@expkey][:exp_last_loaded][:description]
+          @edit[:new_search_name] = @edit[@expkey][:exp_model] != "Storage" ? @edit[:adv_search_name] = @edit[@expkey][:exp_last_loaded][:description] : @edit[:new_search_name]
```

Comment:
in else leg you are assigning same variable (`@edit[:new_search_name]`) ?

```ruby
@edit[:new_search_name] = @edit[:adv_search_name] = @edit[@expkey][:exp_last_loaded][:description] unless @edit[@expkey][:exp_model] == "Storage"
```
does it cover you intention ?